platform-channels.md

```diff
@@ -0,0 +1,249 @@
+enables sending general purpose messages, and `PlatformMethodChannel`
+([API][PlatformMethodChannel]) enables sending messages that correspond to
+method calls.
+
+On the platform side, the class `FlutterMessageChannel` ([Android
+API][FlutterMessageChannelAndroid], [iOS API][FlutterChanneliOS]) enables
+receiving messages from the client, and responding with a result, and the class
+`FlutterMethodChannel` ([Android API][FlutterMethodChannelAndroid], [iOS
+API][FlutterChanneliOS]) enables receiving method calls and sending back a
+result. These classes allow you to develop a platform plugin with very little
+'boilerplate' code.
+
+*Note*: If desired, method calls can also be sent from the host to the client.
+
+[PlatformMessageChannel]: https://docs.flutter.io/flutter/services/PlatformMessageChannel-class.html
+[PlatformMethodChannel]: https://docs.flutter.io/flutter/services/PlatformMethodChannel-class.html
+[FlutterMessageChannelAndroid]: https://docs.flutter.io/javadoc/io/flutter/plugin/common/FlutterMessageChannel.html
+[FlutterMethodChannelAndroid]: https://docs.flutter.io/javadoc/io/flutter/plugin/common/FlutterMethodChannel.html
+[FlutterChanneliOS]: https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/FlutterChannels.mm
+
+### Platform channel data types support and codecs
+
+The standard platform channels use a standard message codec that supports
+efficient binary serialization of simple JSON-like values, such as booleans,
+numbers, Strings, and List and Maps of these (see
+[`StandardMessageCodec`](https://docs.flutter.io/flutter/services/StandardMessageCodec-class.html))
+for details). The serialization and deserialization of these values to and from
+messages happens automatically when you send and receive values.
+
+For more specialized use cases, you may also use the
+[`BinaryCodec`](https://docs.flutter.io/flutter/services/BinaryCodec-class.html),
+[`StringCodec`](https://docs.flutter.io/flutter/services/StringCodec-class.html),
+or
+[`JSONMessageCodec`](https://docs.flutter.io/flutter/services/JSONMessageCodec-class.html).
+
+### Example: Calling platform-specific iOS and Android code using platform channels
+
+The following example demonstrates how to call a new platform-specific API,
+through a concrete example that retrieves and displays the current battery level
+by calling into the Android `BatteryManager` API, and the iOS
+`device.batteryLevel` API, via a single platform message, `getBatteryLevel`.
+
+The standard app template created when you create a new Flutter app can be used.
+
+*Note*: The full source-code for this example is available in
+[`/samples/platform-services/`](https://github.com/flutter/flutter/tree/master/examples/platform_services).
+
+**Step 1: Create the Flutter platform client**
+
+We start by adding support for updating our app's `State` class with the current
+battery state.
+
+First, we construct the channel. Since we need just a single platform message,
```

Comment:
nit: Reverse the sentence, that is start by saying "We use `PlatformMethodChannel` ...."